src/MSBuild/XMake.cs

```diff
@@ -1822,6 +1822,44 @@ private static void GatherResponseFileSwitch(string unquotedCommandLineArg, Comm
             }
         }
 
+        /// <summary>
+        /// Determine if the response file line should be skipped. Lines starting with # should be skipped, except if:
+        ///  - MSBuild is version 16.300 or later
+        ///  - The line starts with a version which is less than the current version.
+        ///     Example: Running MSBuild 16.300, "#v16.300 ..." will be included but "#v16.700 ..." will not.
+        /// </summary>
+        /// <param name="responseFileLine">Response file line. Passed by ref in case it needs to be updated (trimmed version out). </param>
+        /// <returns>True when the line should be skipped. </returns>
+        private static bool SkipResponseFileLine(ref string responseFileLine)
+        {
+            // If it doesn't start with # we should not skip it.
+            if (!responseFileLine.StartsWith("#", StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            // Match "#v16.0 " -> 16.0
+            var match = Regex.Match(responseFileLine, @"^#v(\d+\.\d+) ");
+
+            // If we don't have a version match, skip the file since it started with a #
+            if (!match.Success)
+            {
+                return true;
+            }
+
+            // We have a version in the comment, make sure it's less than or equal to the current version.
+            if (Version.TryParse(match.Groups[1].Value, out Version ver))
+            {
+                if (ver.Major <= ProjectCollection.Version.Major && ver.Minor <= ProjectCollection.Version.Minor)
```

Comment:
This seems wrong, it would reject 17.2 if the actual version is 18.0 because 2 <= 0